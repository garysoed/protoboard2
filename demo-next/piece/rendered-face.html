<style>
  #summary {
    height: 4rem;
    width: 4rem;
  }
</style>
<pb-lens id="lens">
  <div id="summary">
    <mk-icon id="icon" fit-to="height"></mk-icon>
  </div>
</pb-lens>