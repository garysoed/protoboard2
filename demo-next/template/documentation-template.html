<style>
  #content::slotted(*) {
    padding-left: 1rem;
    padding-right: 1rem;
  }

  #title {
    padding-left: 1rem;
    padding-right: 1rem;
  }
</style>

<section flex>
  <h3 id="title" mk-display></h3>
  <slot id="content" name="content"></slot>
</section>
