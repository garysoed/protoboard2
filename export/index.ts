export {Lens, $lens} from '../src/core/lens';
export {LensDisplay} from '../src/core/lens-display';

// action
export {pickAction} from '../src/action/pick-action';

// coordinate
export {Cartesian} from '../src/coordinate/cartesian';
export {Coordinate} from '../src/coordinate/coordinate';
export {Indexed} from '../src/coordinate/indexed';

// region
export {Deck, $deck, DeckSpec} from '../src/region/deck';
export {Slot, $slot, SlotSpec, slotSpec} from '../src/region/slot';

// core
export {$createSpecEntries, $createSpecMap} from '../src/objects/object-create-spec';
export {ContentSpec} from '../src/payload/is-container';
export {ObjectSpec} from '../src/types/object-spec';
export {$render, Render} from '../src/objects/render';
export {$active, Active, activeSpec} from '../src/core/active';
export {$$activeSpec} from '../src/objects/active-spec';

// piece
export {$d1, D1, D1Spec, d1Spec} from '../src/piece/d1';
export {$d2, D2, D2Spec} from '../src/piece/d2';
export {$d6, D6, D6Spec} from '../src/piece/d6';

// tiling
export {rectOrthogonal, RectOrthogonal, Direction as RectOrthogonalDirection, distance as rectOrthogonalDistance} from '../src/tiling/rect-orthogonal';
export {vhex, VHex, Direction as VHexDirection, distance as vhexDistance} from '../src/tiling/vhex';
export {Tile} from '../src/tiling/types';
