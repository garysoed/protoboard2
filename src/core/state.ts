import { Observable } from 'rxjs';

/**
 * State of a game during its runtime.
 *
 * @thModule core
 */
export interface State {
  /**
   * Identifies the object. Must be globally unique.
   */
  readonly id: string;

  /**
   * Used to map to a function that generates the object.
   */
  readonly type: string;

  /**
   * Used to generate the object.
   */
  readonly payload: ReadonlyMap<string, Observable<unknown>>;
}
