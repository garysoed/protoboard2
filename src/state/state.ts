import { BehaviorSubject } from 'rxjs';

/**
 * State of a game during its runtime.
 *
 * @thModule core
 */
export interface State<P extends object> {
  /**
   * Identifies the object. Must be globally unique.
   */
  readonly id: string;

  /**
   * Used to map to a function that generates the object.
   */
  readonly type: string;

  /**
   * Used to generate the object.
   * TODO: Instead of BehaviorSubject, this should be split into Observable and Subject/
   */
  readonly payload: {readonly [K in keyof P]: BehaviorSubject<P[K]>};
}
