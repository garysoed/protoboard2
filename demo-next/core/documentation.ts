import {cache} from 'gs-tools/export/data';
import {renderTheme} from 'mask';
import {Context, Ctrl, DIV, ocase, query, registerCustomElement, renderElement, RenderSpec} from 'persona';
import {Observable} from 'rxjs';
import {map} from 'rxjs/operators';

import {D1_DEMO} from '../piece/d1';
import {D2_DEMO} from '../piece/d2';
import {D6_DEMO} from '../piece/d6';

import template from './documentation.html';
import {INSTRUCTION} from './instruction';
import {$locationService} from './location-service';
import {getPageSpec, PageSpec} from './page-spec';


const $documentation = {
  shadow: {
    root: query('#root', DIV, {
      content: ocase<PageSpec|null>('#content'),
    }),
  },
};

export class Documentation implements Ctrl {
  constructor(private readonly $: Context<typeof $documentation>) { }

  @cache()
  get runs(): ReadonlyArray<Observable<unknown>> {
    return [
      renderTheme(this.$),
      $locationService.get(this.$.vine).location$.pipe(map(location => getPageSpec(location.type))).pipe(
          this.$.shadow.root.content(spec => this.renderContent(spec)),
      ),
    ];
  }

  @cache()
  private renderContent(spec: PageSpec|null): RenderSpec|null {
    if (!spec) {
      return null;
    }

    return renderElement({spec: {}, registration: spec.registration});
  }
}

export const DOCUMENTATION = registerCustomElement({
  ctrl: Documentation,
  deps: [
    // CanvasDemo,
    D1_DEMO,
    D2_DEMO,
    D6_DEMO,
    // DeckDemo,
    INSTRUCTION,
    // SlotDemo,
  ],
  spec: $documentation,
  tag: 'pbd-documentation',
  template,
});