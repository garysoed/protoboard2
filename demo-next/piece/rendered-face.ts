import {cache} from 'gs-tools/export/data';
import {ICON, registerSvg, renderTheme} from 'mask';
import {Context, Ctrl, H3, iattr, id, otext, P, registerCustomElement} from 'persona';
import {Observable} from 'rxjs';
import {map} from 'rxjs/operators';

import cardBackSvg from '../asset/card_back.svg';
import cardFrontSvg from '../asset/card_front.svg';
import coinFrontSvg from '../asset/coin.svg';
import coinBackSvg from '../asset/coin_back.svg';
import dicePip1Svg from '../asset/dice_pip_1.svg';
import dicePip2Svg from '../asset/dice_pip_2.svg';
import dicePip3Svg from '../asset/dice_pip_3.svg';
import dicePip4Svg from '../asset/dice_pip_4.svg';
import dicePip5Svg from '../asset/dice_pip_5.svg';
import dicePip6Svg from '../asset/dice_pip_6.svg';
import gemSvg from '../asset/gem.svg';
import meepleSvg from '../asset/meeple.svg';

import template from './rendered-face.html';


export enum FaceType {
  CARD_BACK = 'cardback',
  CARD_FRONT = 'cardfront',
  COIN_BACK = 'coinback',
  COIN_FRONT = 'coinfront',
  DICE_PIP_1 = 'dicepip1',
  DICE_PIP_2 = 'dicepip2',
  DICE_PIP_3 = 'dicepip3',
  DICE_PIP_4 = 'dicepip4',
  DICE_PIP_5 = 'dicepip5',
  DICE_PIP_6 = 'dicepip6',
  GEM = 'gem',
  MEEPLE = 'meeple',
}

const $renderedFace = {
  host: {
    faceType: iattr('face-type'),
  },
  shadow: {
    icon: id('icon', ICON),
    name: id('name', H3, {
      text: otext(),
    }),
    details: id('details', P),
  },
};


class RenderedFace implements Ctrl {
  constructor(private readonly $: Context<typeof $renderedFace>) { }

  get runs(): ReadonlyArray<Observable<unknown>> {
    return [
      renderTheme(this.$),
      this.icon$.pipe(this.$.shadow.icon.icon()),
      this.faceName$.pipe(this.$.shadow.name.text()),
    ];
  }

  @cache()
  private get icon$(): Observable<string> {
    return this.$.host.faceType.pipe(map(faceType => faceType ?? ''));
  }

  @cache()
  private get faceName$(): Observable<string> {
    return this.$.host.faceType.pipe(
        map(faceType => {
          switch (faceType) {
            case FaceType.CARD_BACK:
              return 'Card Back';
            case FaceType.CARD_FRONT:
              return 'Card Front';
            case FaceType.COIN_BACK:
              return 'Coin Back';
            case FaceType.COIN_FRONT:
              return 'Coin Front';
            case FaceType.DICE_PIP_1:
              return 'Dice 1 (pip)';
            case FaceType.DICE_PIP_2:
              return 'Dice 2 (pip)';
            case FaceType.DICE_PIP_3:
              return 'Dice 3 (pip)';
            case FaceType.DICE_PIP_4:
              return 'Dice 4 (pip)';
            case FaceType.DICE_PIP_5:
              return 'Dice 5 (pip)';
            case FaceType.DICE_PIP_6:
              return 'Dice 6 (pip)';
            case FaceType.GEM:
              return 'Gem';
            case FaceType.MEEPLE:
              return 'Meeple';
            default:
              return '';
          }
        }),
    );
  }
}

export const RENDERED_FACE = registerCustomElement({
  configure: vine => {
    registerSvg(vine, FaceType.CARD_BACK, {type: 'embed', content: cardBackSvg});
    registerSvg(vine, FaceType.CARD_FRONT, {type: 'embed', content: cardFrontSvg});
    registerSvg(vine, FaceType.COIN_BACK, {type: 'embed', content: coinBackSvg});
    registerSvg(vine, FaceType.COIN_FRONT, {type: 'embed', content: coinFrontSvg});
    registerSvg(vine, FaceType.GEM, {type: 'embed', content: gemSvg});
    registerSvg(vine, FaceType.MEEPLE, {type: 'embed', content: meepleSvg});
    registerSvg(vine, FaceType.DICE_PIP_1, {type: 'embed', content: dicePip1Svg});
    registerSvg(vine, FaceType.DICE_PIP_2, {type: 'embed', content: dicePip2Svg});
    registerSvg(vine, FaceType.DICE_PIP_3, {type: 'embed', content: dicePip3Svg});
    registerSvg(vine, FaceType.DICE_PIP_4, {type: 'embed', content: dicePip4Svg});
    registerSvg(vine, FaceType.DICE_PIP_5, {type: 'embed', content: dicePip5Svg});
    registerSvg(vine, FaceType.DICE_PIP_6, {type: 'embed', content: dicePip6Svg});
  },
  ctrl: RenderedFace,
  deps: [
    ICON,
  ],
  spec: $renderedFace,
  tag: 'pbd-rendered-face',
  template,
});