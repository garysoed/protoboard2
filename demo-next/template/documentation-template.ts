import {cache} from 'gs-tools/export/data';
import {mapNullableTo} from 'gs-tools/export/rxjs';
import {renderTheme} from 'mask';
import {Context, Ctrl, H3, iattr, id, otext, registerCustomElement} from 'persona';
import {Observable} from 'rxjs';

import template from './documentation-template.html';


const $documentationTemplate = {
  host: {
    label: iattr('label'),
  },
  shadow: {
    title: id('title', H3, {
      text: otext(),
    }),
  },
};

class DocumentationTemplate implements Ctrl {
  constructor(private readonly $: Context<typeof $documentationTemplate>) { }

  @cache()
  get runs(): ReadonlyArray<Observable<unknown>> {
    return [
      renderTheme(this.$),
      this.$.host.label.pipe(mapNullableTo(''), this.$.shadow.title.text()),
    ];
  }
}

export const DOCUMENTATION_TEMPLATE = registerCustomElement({
  ctrl: DocumentationTemplate,
  spec: $documentationTemplate,
  tag: 'pbd-documentation-template',
  template,
});
