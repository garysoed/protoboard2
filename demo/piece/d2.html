<pbd-piece-template id="template" label="D2" face-count="2" component-tag="pb-d2">
  <p mk-body-1 slot="details">
    A D2 is the most is a piece with 2 faces. Some examples include coins and cards.
  </p>
</pbd-piece-template>
