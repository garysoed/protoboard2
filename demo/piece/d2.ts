import {immutablePathSource, rootStateIdSource} from 'grapevine';
import {mutableState} from 'gs-tools/export/state';
import {BaseThemedCtrl, Icon, _p} from 'mask';
import {element, PersonaContext} from 'persona';
import {Observable, of} from 'rxjs';

import {$registerRenderObject, $slot, D2, D2Spec, d2Spec, Lens, Slot, slotSpec, SlotSpec} from '../../export';
import {FaceType, RenderedFace} from '../core/rendered-face';
import {DocumentationTemplate} from '../template/documentation-template';

import template from './d2.html';
import {renderPiece} from './render-piece';


interface State {
  readonly cardSlot: SlotSpec;
  readonly coinSlot: SlotSpec;
}

const $cardId = rootStateIdSource<D2Spec>(() => d2Spec({}));
const $cardPath = immutablePathSource($cardId);
const $coinId = rootStateIdSource<D2Spec>(() => d2Spec({}));
const $coinPath = immutablePathSource($coinId);

const $stateId = rootStateIdSource<State>(vine => ({
  cardSlot: slotSpec({contentsId: mutableState([$cardPath.get(vine)])}),
  coinSlot: slotSpec({contentsId: mutableState([$coinPath.get(vine)])}),
}));

const $cardSlotPath = immutablePathSource($stateId, state => state._('cardSlot'));
const $coinSlotPath = immutablePathSource($stateId, state => state._('coinSlot'));


export const $d2Demo = {
  tag: 'pbd-d2',
  api: {},
};

const $ = {
  cardSlot: element('cardSlot', $slot, {}),
  coinSlot: element('coinSlot', $slot, {}),
};

@_p.customElement({
  ...$d2Demo,
  configure: vine => {
    const cardPath = $cardPath.get(vine);
    const coinPath = $coinPath.get(vine);
    const registerRenderObject = $registerRenderObject.get(vine);
    registerRenderObject(
        cardPath,
        renderPiece([FaceType.CARD_BACK, FaceType.CARD_FRONT], cardPath),
    );

    registerRenderObject(
        coinPath,
        renderPiece([FaceType.COIN_FRONT, FaceType.COIN_BACK], coinPath),
    );
  },
  dependencies: [
    D2,
    DocumentationTemplate,
    Icon,
    Lens,
    RenderedFace,
    Slot,
  ],
  template,
})
export class D2Demo extends BaseThemedCtrl<typeof $> {
  constructor(context: PersonaContext) {
    super(context, $);
  }

  protected get renders(): ReadonlyArray<Observable<unknown>> {
    return [
      this.renderers.cardSlot.objectPath(of($cardSlotPath.get(this.vine))),
      this.renderers.coinSlot.objectPath(of($coinSlotPath.get(this.vine))),
    ];
  }
}
