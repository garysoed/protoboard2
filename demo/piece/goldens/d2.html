<pbd-d2><style>
  .slotContainer {
    display: inline-block;
    position: relative;
    width: 12rem;
    height: 12rem;
  }
</style>

<pbd-documentation-template mk-body-1="" label="D2"><style>
  #content::slotted(*) {
    padding-left: 1rem;
    padding-right: 1rem;
  }

  #title {
    padding-left: 1rem;
    padding-right: 1rem;
  }
</style>

<section flex="">
  <h3 id="title" mk-display="">D2</h3>
  <slot id="content" name="content"><div slot="content">
    <p>
      A D2 is a piece with 2 faces.
    </p>
    <p>
      Some examples:
    </p>
    <ul>
      <li>
        <p>Card</p>
        <div class="slotContainer">
          <pb-slot id="cardSlot" object-path="i::6"><style>
  #root {
    align-items: center;
    justify-content: center;
    background-color: var(--mkThemeBG);
    border: 1px solid var(--mkThemeFG);
    box-sizing: border-box;
    flex-wrap: wrap;
  }
</style>

<div id="root" fill-parent="" mk-action-2="" layout="row">
  <!-- #content--><pb-d2 object-path="i::2"><slot id="face" name="face-0"><pbd-rendered-face face-type="cardback" slot="face-0"><style>
  #summary {
    height: 4rem;
    width: 4rem;
  }
</style>
<pb-lens><style>
  #details {
    display: none;
  }
</style>

<div id="root">
  <slot name="summary"><div id="summary" slot="summary">
    <mk-icon id="icon" aria-hidden="true" role="presentation" icon="cardback"><style>
  :host {
    pointer-events: none;
    user-select: none;
  }

  #root > svg #foreground {
    fill: var(--mkThemeFG) !important;
  }
</style>
<span id="root">
  <!-- #content --><svg xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:cc="http://creativecommons.org/ns#" xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#" xmlns:svg="http://www.w3.org/2000/svg" xmlns="http://www.w3.org/2000/svg" xmlns:sodipodi="http://sodipodi.sourceforge.net/DTD/sodipodi-0.dtd" xmlns:inkscape="http://www.inkscape.org/namespaces/inkscape" inkscape:version="1.0 (4035a4f, 2020-05-01)" sodipodi:docname="card_back.svg" id="svg22" y="0px" x="0px" version="1.1" viewBox="0 0 24 24" height="100%">
  <metadata id="metadata28">
    <rdf:RDF>
      <cc:Work rdf:about="">
        <dc:format>image/svg+xml</dc:format>
        <dc:type rdf:resource="http://purl.org/dc/dcmitype/StillImage"></dc:type>
        <dc:title>King of Spades inverted</dc:title>
      </cc:Work>
    </rdf:RDF>
  </metadata>
  <defs id="defs26"></defs>
  <sodipodi:namedview inkscape:current-layer="svg22" inkscape:window-maximized="0" inkscape:window-y="286" inkscape:window-x="-13" inkscape:cy="51.525298" inkscape:cx="-21.800595" inkscape:zoom="5.376" showgrid="false" id="namedview24" inkscape:window-height="870" inkscape:window-width="1381" inkscape:pageshadow="2" inkscape:pageopacity="0" guidetolerance="10" gridtolerance="10" objecttolerance="10" borderopacity="1" inkscape:document-rotation="0" bordercolor="#666666" pagecolor="#ffffff" units="px"></sodipodi:namedview>
  <title id="title10">King of Spades inverted</title>
  <desc id="desc12">Created with Sketch.</desc>
  <path style="stroke:none;stroke-width:0.266667" d="M 5.3333335,0 H 18.666666 c 0.662742,0 1.2,0.5372583 1.2,1.2 v 21.6 c 0,0.662742 -0.537258,1.2 -1.2,1.2 H 5.3333335 c -0.662742,0 -1.2,-0.537258 -1.2,-1.2 V 1.2 c 0,-0.6627417 0.537258,-1.2 1.2,-1.2 z" id="foreground"></path>
  <text id="text18" font-family="'Helvetica Neue', Helvetica, Arial-Unicode, Arial, Sans-serif" font-weight="bold" font-size="5px" fill="#000000" y="124.11459" x="0.18601191">Created by Joel Wisneski</text>
  <text id="text20" font-family="'Helvetica Neue', Helvetica, Arial-Unicode, Arial, Sans-serif" font-weight="bold" font-size="5px" fill="#000000" y="129.11458" x="0.18601191">from the Noun Project</text>
</svg>
</span>
<link rel="stylesheet" href="base/test.css"></mk-icon>
  </div></slot>
</div>
<slot id="details" name="details"><div slot="details">
    <h3 id="name">Card Back</h3>
    <p mk-body-1="">More detailed information on the piece goes here.</p>
  </div></slot><link rel="stylesheet" href="base/test.css"></pb-lens><link rel="stylesheet" href="base/test.css"></pbd-rendered-face></slot>
<link rel="stylesheet" href="base/test.css"></pb-d2>
</div>
<link rel="stylesheet" href="base/test.css"></pb-slot>
        </div>
      </li>
      <li>
        <p>Coin</p>
        <div class="slotContainer">
          <pb-slot id="coinSlot" object-path="i::7"><style>
  #root {
    align-items: center;
    justify-content: center;
    background-color: var(--mkThemeBG);
    border: 1px solid var(--mkThemeFG);
    box-sizing: border-box;
    flex-wrap: wrap;
  }
</style>

<div id="root" fill-parent="" mk-action-2="" layout="row">
  <!-- #content--><pb-d2 object-path="i::4"><slot id="face" name="face-0"><pbd-rendered-face face-type="coinfront" slot="face-0"><style>
  #summary {
    height: 4rem;
    width: 4rem;
  }
</style>
<pb-lens><style>
  #details {
    display: none;
  }
</style>

<div id="root">
  <slot name="summary"><div id="summary" slot="summary">
    <mk-icon id="icon" aria-hidden="true" role="presentation" icon="coinfront"><style>
  :host {
    pointer-events: none;
    user-select: none;
  }

  #root > svg #foreground {
    fill: var(--mkThemeFG) !important;
  }
</style>
<span id="root">
  <!-- #content --><svg xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:cc="http://creativecommons.org/ns#" xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#" xmlns:svg="http://www.w3.org/2000/svg" xmlns="http://www.w3.org/2000/svg" xmlns:sodipodi="http://sodipodi.sourceforge.net/DTD/sodipodi-0.dtd" xmlns:inkscape="http://www.inkscape.org/namespaces/inkscape" sodipodi:docname="coin.svg" inkscape:version="1.0 (4035a4f, 2020-05-01)" id="svg4170" xml:space="preserve" enable-background="new 0 0 100 100" viewBox="0 0 24 24" y="0px" x="0px" version="1.1" fill="#000000" height="100%"><metadata id="metadata4178"><rdf:RDF><cc:Work rdf:about=""><dc:format>image/svg+xml</dc:format><dc:type rdf:resource="http://purl.org/dc/dcmitype/StillImage"></dc:type><dc:title></dc:title></cc:Work></rdf:RDF></metadata><defs id="defs4176"></defs><sodipodi:namedview inkscape:document-rotation="0" inkscape:current-layer="svg4170" inkscape:window-maximized="0" inkscape:window-y="23" inkscape:window-x="0" inkscape:cy="40.380379" inkscape:cx="2.8697312" inkscape:zoom="3.095" showgrid="false" id="namedview4174" inkscape:window-height="825" inkscape:window-width="1393" inkscape:pageshadow="2" inkscape:pageopacity="0" guidetolerance="10" gridtolerance="10" objecttolerance="10" borderopacity="1" bordercolor="#666666" pagecolor="#ffffff"></sodipodi:namedview><path inkscape:connector-curvature="0" id="foreground" d="M 12,0 A 12,12 0 0 0 0,12 12,12 0 0 0 12,24 12,12 0 0 0 24,12 12,12 0 0 0 12,0 Z m -0.435,3.9796874 h 1.258125 v 1.8576564 c 0.945536,0.1294484 1.680172,0.5260299 2.203594,1.1901562 0.52905,0.6641261 0.793594,1.5729398 0.793594,2.7267187 h -1.553438 c 0,-0.7935746 -0.188692,-1.4237981 -0.565781,-1.8909374 C 13.324005,7.3961414 12.814582,7.1625 12.172969,7.1625 c -0.669755,0 -1.181987,0.1746463 -1.536563,0.5235938 -0.354576,0.3433196 -0.531562,0.8246494 -0.531562,1.44375 0,0.5740752 0.185414,1.0324622 0.556875,1.3757812 0.377088,0.337691 0.993734,0.647184 1.849218,0.928594 0.861113,0.275781 1.530667,0.582465 2.009063,0.920156 0.484024,0.332063 0.838466,0.720686 1.063594,1.165312 0.230756,0.444628 0.346406,0.965288 0.346406,1.561876 0,0.951163 -0.287018,1.716234 -0.861094,2.295937 -0.568447,0.579703 -1.367697,0.923377 -2.397656,1.030313 v 1.6125 h -1.249687 v -1.6125 C 10.37472,18.312133 9.5529943,17.940367 8.9564062,17.293125 8.3654465,16.640255 8.07,15.756725 8.07,14.642344 h 1.5618749 c 0,0.782318 0.2055485,1.384448 0.6164061,1.806562 0.410858,0.422115 0.993323,0.633281 1.7475,0.633281 0.737293,0 1.316948,-0.177455 1.739063,-0.532031 0.422114,-0.354576 0.633281,-0.838246 0.633281,-1.451719 0,-0.517793 -0.183074,-0.954174 -0.548906,-1.30875 C 13.459015,13.435111 12.848456,13.114381 11.987344,12.827344 10.78854,12.461511 9.9134376,11.9858 9.3618749,11.400469 8.8159409,10.809509 8.5429687,10.061294 8.5429687,9.1551562 8.5429687,8.2321334 8.810322,7.4750222 9.345,6.8840626 9.885306,6.2931026 10.625093,5.94147 11.565,5.8289062 Z" style="color:#000000;clip-rule:nonzero;display:inline;overflow:visible;visibility:visible;opacity:1;isolation:auto;mix-blend-mode:normal;color-interpolation:sRGB;color-interpolation-filters:linearRGB;solid-color:#000000;solid-opacity:1;fill:#000000;fill-opacity:1;fill-rule:nonzero;stroke:none;stroke-width:2.4;stroke-linecap:butt;stroke-linejoin:miter;stroke-miterlimit:4;stroke-dasharray:none;stroke-dashoffset:0;stroke-opacity:1;color-rendering:auto;image-rendering:auto;shape-rendering:auto;text-rendering:auto;enable-background:accumulate"></path></svg>
</span>
<link rel="stylesheet" href="base/test.css"></mk-icon>
  </div></slot>
</div>
<slot id="details" name="details"><div slot="details">
    <h3 id="name">Coin Front</h3>
    <p mk-body-1="">More detailed information on the piece goes here.</p>
  </div></slot><link rel="stylesheet" href="base/test.css"></pb-lens><link rel="stylesheet" href="base/test.css"></pbd-rendered-face></slot>
<link rel="stylesheet" href="base/test.css"></pb-d2>
</div>
<link rel="stylesheet" href="base/test.css"></pb-slot>
        </div>
      </li>
    </ul>
  </div></slot>
</section>
<link rel="stylesheet" href="base/test.css"></pbd-documentation-template>
<link rel="stylesheet" href="base/test.css"></pbd-d2>